src/soap/serializers/site.js

```diff
@@ -16,32 +16,87 @@ export function siteInfoEnd() {
   return '</siteInfo>'
 }
 
-export function siteInfoType({ station }) {
+export function siteInfoType({ externalRef, station }) {
+  const siteCode =
+    externalRef && externalRef.siteCode && externalRef.siteCode.split('-')[1]
   return (
     `<siteName>${encodeXML(station.name)}</siteName>` +
-    `<siteCode network="${encodeXML(
-      (station.organization_lookup && station.organization_lookup.slug) ||
-        'dendra'
-    )}">${encodeXML(station._id)}</siteCode>` +
+    (externalRef && externalRef.siteCode
+      ? `<siteCode network="${encodeXML(
+          (station.organization_lookup && station.organization_lookup.slug) ||
+            'dendra'
+        )}" siteID="${externalRef.siteId}">${encodeXML(
+          externalRef.siteCode
+        )}</siteCode>`
+      : '') +
     (station.geo && station.geo.type === 'Point'
       ? '<geoLocation>' +
-        `<geogLocation xsi:type="LatLonPointType">` +
+        `<geogLocation xsi:type="LatLonPointType" ${
+          siteCode !== 'full' ? 'srs="EPSG:99999"' : ''
+        }>` +
         `<latitude>${encodeXML(station.geo.coordinates[1] + '')}</latitude>` +
         `<longitude>${encodeXML(station.geo.coordinates[0] + '')}</longitude>` +
         '</geogLocation>' +
+        `${
+          siteCode === 'full'
+            ? // (externalRef.localX || externalRef.localY)
+              `<localSiteXY projectionInformation="WGS 84 / UTM zone 19N">${
+                externalRef.localX
+                  ? `<X>${encodeXML(externalRef.localX)}</X>`
+                  : ''
+              }${
+                externalRef.localY
+                  ? `<Y>${encodeXML(externalRef.localY)}</Y>`
+                  : ''
+              }</localSiteXY>`
+            : ''
+        }` +
         '</geoLocation>'
       : '') +
-    (station.geo &&
+    (siteCode === 'full' &&
```

Comment:
We should never hardcode a siteCode. There are thousands of them and they are defined externally in databases.